product_subscription/models/subscription_object.py

```diff
@@ -65,6 +70,26 @@ def _compute_renewed(self):
             )
             subscription.renewed = True if subscriptions else False
 
+    @api.multi
+    @api.depends("state", "is_trial", "subscribed_on", "subscriber")
+    def _compute_is_first_subscription(self):
+        for subscription in self:
+            if subscription.is_trial or subscription.state in (
+                "draft",
+                "cancel",
+            ):
+                subscription.is_first_subscription = False
```

Comment:
I don't why to loop on this ones to set their field to False as it's a boolean... If you don't process them they'll have the expected value... so a filter on the self should be enough to exclude them from the loop.

Remember that until v13 this subscription.is_first_subscription = False send at least an update sql query to the database.

Reply:
Indeed but I needed to store `is_first_subscription` (cf fixup commits) so I need to loop on all (triggered) subscriptions to set `is_first_subscription` to False in case `state` or `is_trial` changes.